dom0-updates/qfile-dom0-unpacker.c

```diff
@@ -49,13 +53,22 @@ int main(int argc, char ** argv)
 	long long files_limit = DEFAULT_MAX_UPDATES_FILES;
 	long long bytes_limit = DEFAULT_MAX_UPDATES_BYTES;
 
+	struct statvfs st;
+	long long root_free_space;
+
 	if (argc < 3) {
 		fprintf(stderr, "Invalid parameters, usage: %s user dir [-v]\n", argv[0]);
 		exit(1);
 	}
 
+	statvfs("/", &st);
+	// take a little margin with 90% of the free space
+	root_free_space = max(0, st.f_bfree * st.f_frsize * 0.90);
+
+	bytes_limit = min(root_free_space, DEFAULT_MAX_UPDATES_BYTES);
+
 	if ((var=getenv("UPDATES_MAX_BYTES")))
-		bytes_limit = atoll(var);
+		bytes_limit = min(root_free_space, atoll(var));
```

Comment:
Explicit env variable should override the limit. This is specifically used during backup restore, where the backup is processed in parallel, so it will attempt to free the space in the meantime (and if we port https://github.com/QubesOS/qubes-core-agent-linux/pull/184/commits/7c261f45da20f7a57c18eff202d2d9dc07804e30 to dom0 version too, it's going to be guaranteed)